airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowTest.java

```diff
@@ -1013,6 +1028,185 @@ void testGetStuckAfterRun(Consumer<ConnectionManagerWorkflow> signalSender, Thre
 
   }
 
+  @Nested
+  @DisplayName("Test workflow where the child workflow throw a hearbeat timeout exception")
+  class HeartbeatFailureWorkflow {
+
+    @BeforeEach
+    public void setup() {
+      setupSpecificChildWorkflow(SyncWorkflowFailingWithHearbeatTimeoutException.class);
+    }
+
+    public static Stream<Arguments> getMaxAttemptForResetRetry() {
+      return Stream.of(
+          Arguments.of(3), // "The max attempt is 3, it will test that after a failed attempt the next attempt will also be a
+                           // reset")
+          Arguments.of(1) // "The max attempt is 3, it will test that after a failed job the next attempt will also be a job")
```

Comment:
should this say `reset` instead of `job`?

Reply:
I would prefer to keep it as `job` since it is describe the job that we have in the configDB

Reply:
sorry I should be clearer, I meant `reset job` instead of just `job`.

Reply:
Ahh make sense, Done